test/e2e/framework/ginkgowrapper.go

```diff
@@ -123,6 +124,26 @@ func AnnotatedLocationWithOffset(annotation string, offset int) types.CodeLocati
 	return codeLocation
 }
 
+// SIGDescribe returns a wrapper function for ginkgo.Describe which injects
+// the SIG name as annotation. The parameter should be lowercase with
+// no spaces and no sig- or SIG- prefix.
+func SIGDescribe(sig string) func(string, ...interface{}) bool {
+	if !sigRE.MatchString(sig) || strings.HasPrefix(sig, "sig-") {
+		panic(fmt.Sprintf("SIG label must be lowercase, no spaces and no sig- prefix, got instead: %q", sig))
+	}
```

Comment:
this is a strong convention and panicing may cause some developer bad UX, because it will be hard to identify the problem ... a linter perhaps?

Reply:
You got me here :sweat_smile: Doing the validation inline with panic is trivial, so I went for that first. But you are right, the panic and the root cause for it are not obvious. My plan (not fully worked out yet) was to have an invocation of the test suites as part of `pull-kubernetes-verify` which as clearly as possible shows what problems exist in the suite.

However, writing a linter that handles all cases will not be possible and a lot of work. What it can't handle are calls where the strings are computed and only known at runtime.

Reply:
No more panics...

Reply:
ok, found the original comment hahah, this is from Jul, my memory has shorter buffer XD